### PR TITLE
CLOUDP-61387: Evergreen packaging improvements

### LINKED ISSUE #130: Debian package installs into /usr/local
I downloaded and installed the Debian package of mongocli, and it had the following files:

<pre>
carbon% dpkg -L mongocli                 
/usr
/usr/local
/usr/local/bin
/usr/local/bin/mongocli
</pre>

Per [Debian documentation](https://www.debian.org/doc/manuals/maint-guide/modify.en.html#destdir) the packaged software is supposed to use `/usr` rather than `/usr/local` namespace.

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

- Improve naming to match what our repos do currently
- improve how we use curator to handle versions as the server does
- closes #130 

New package name for rpm
`mongodb-cli-unstable-0.2.0.latest.x86_64.rpm`
Example name from server
`mongodb-org-unstable-4.3.6-0.1.latest.el7.x86_64.rpm`

New package name for deb
`mongodb-cli-unstable_0.2.0~latest_amd64.deb`
Example name from server
`mongodb-org-unstable_4.3.6~latest_amd64.deb`

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -42,34 +39,12 @@ brews:
     system "#{bin}/mongocli --version"
   install: |
     bin.install "mongocli"
-nfpms:
```
Comment (gssbzn) [inline .goreleaser.yml:45]: Extracting this to a new file

File: .goreleaser_evg.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+before:
```
Comment (gssbzn) [inline .goreleaser_evg.yml:1]: let EVG have its own file to simplify naming startegy and skip things we don't need on evg (brew)

**Comment by gssbzn** [comment]
@tychoish this is mostly setting up things to improve our naming and use a project setting for version, we may need to work on how to set that up, would be great if evg allowed tasks to un on tag

File: .goreleaser.yml [hunk 3]
```diff
@@ -31,46 +33,31 @@ brews:
   github:
     owner: mongodb-labs
     name: homebrew-mongocli
-  url_template: "https://github.com/mongodb/mongocli/releases/download/{{ .Tag }}/{{ .ArtifactName }}"
-  commit_author:
-    name: gustavo.bazan
-    email: gustavo.bazan@mongodb.com
-  homepage: "https://github.com/mongodb/mongocli"
+  skip_upload: auto
   description: The MongoDB Command Line Interface (mongocli) is a tool for managing your MongoDB cloud services, like MongoDB Atlas, MongoDB Cloud Manager, and MongoDB Ops Manager.
```
Comment (colm-quinn) [inline .goreleaser.yml:37]: [q] Are these descriptions still needed as you've moved them into `.goreleaseer_evg.yml`? Just wondering can we reduce the reuse of magic string and handle it all in the linked script.

Comment (gssbzn) [reply .goreleaser.yml:37]: yes but I decided against this, the whole point was about naming and I think I going to do this in evg

**Comment by gssbzn** [comment]
@colm-quinn decided to rollback the two files change and add a step in evergreen to manage the naming for yum/apt
Waiting for @tychoish on curator changes

**Comment by gssbzn** [comment]
@tychoish I'm going to merge this, please feel free to add any comment if you feel we are missing anything and we'll address this on a separate PR

**Pull request merged**